src/app/app.component.ts

```diff
@@ -198,35 +207,84 @@ export class AppComponent implements AfterViewInit {
 
   async initializeTreeAsync() {
     this.ref.detectChanges();
-    await this.lfFolderBrowserElement?.nativeElement.initAsync({
-      treeService: this.lfRepoTreeService
-    });
+    let defaultNode = {
+      id: '168158',
+      isContainer: true,
+      isLeaf: false,
+      path: '\\Ke\\test\\new folder test',
+      name: 'new folder test',
+      icon: ''
+    };
+    let focusedNode;
+    if (this.selectedFolder) {
+      focusedNode = {
+        id: this.selectedFolder.entryId,
+        isContainer: true,
+        isLeaf: false,
+        path: this.selectedFolder.path,
+        name: this.selectedFolder.displayName,
+        icon: ''
+      }
+    } else {
+      focusedNode = defaultNode;
+    }
+    await this.lfRepositoryBrowser?.nativeElement.initAsync(this.lfRepoTreeNodeService, focusedNode);
+  }
+
+  isNodeSelectable = (node: LfRepoTreeNode) => {
+    if (node.entryType == EntryType.Folder) {
+      return true
+    }
+    else {
+      return false
+    }
   }
 
   get isLoggedIn(): boolean {
     return this.loginComponent?.nativeElement?.state === LoginState.LoggedIn;
   }
 
   // Tree event handler methods
-  async onOkClick(okClickEvent: Event) {
-    const selectedNode = (okClickEvent as CustomEvent<TreeNode>).detail;
-    const breadcrumbs = this.lfFolderBrowserElement?.nativeElement?.breadcrumbs;
-    const entryId = Number.parseInt(selectedNode.id, 10);
-    const path = selectedNode.path;
-    this.selectedFolder = {
-      entryId,
-      path,
-      displayName: this.getFolderNameText(entryId, path),
-      displayPath: this.getFolderPathTooltip(path)
-    };
-    if (breadcrumbs) {
-      this.selectedFolder.breadcrumbs = breadcrumbs;
-    }
-    const nodeId = selectedNode.id;
-    const repoId = (await this.repoClient.getCurrentRepoId());
-    const waUrl = this.loginComponent.nativeElement.account_endpoints.webClientUrl;
-    this.selectedNodeUrl = getEntryWebAccessUrl(nodeId, repoId, waUrl, selectedNode.isContainer);
-    this.expandFolderBrowser = false;
+  async onSelectFolder() {
+    const selectedNode = this.lfRepositoryBrowser.nativeElement.currentFolder as LfRepoTreeNode;
+
+
+      const breadcrumbs = this.lfRepositoryBrowser?.nativeElement?.breadcrumbs;
+      let entryId = Number.parseInt(selectedNode.id, 10);
+      const path = selectedNode.path;
+      if (selectedNode.entryType == EntryType.Shortcut) {
+        entryId = selectedNode.targetId;
+      }
+      this.selectedFolder = {
```

Comment:
maybe instead of storing the folder, it should actually just store the path and call getEntry ByPath when initializing the tree